controllers/drcluster_controller.go

```diff
@@ -238,32 +278,294 @@ func (u *drclusterUpdater) finalizerRemove() error {
+			"NeworkFence ManifestWork for unfence failed")
+
+		return true, fmt.Errorf("failed to generate NetworkFence MW on cluster %s to unfence %s",
+			peerCluster.Name, u.object.Name)
+	}
+
+	setDRClusterUnfencedCondition(&u.object.Status.Conditions, u.object.Generation,
+		"Cluster successfully unfenced")
+
+	return false, nil
+}
+
+// We are here means following things have been confirmed.
+// 1) Fencing CR MCV was obtained.
+// 2) MCV for the Fencing CR showed the cluster as unfenced
+//
+// * Proceed to delete the ManifestWork for the fencingCR
+// * Issue a requeue
+func (u *drclusterUpdater) cleanCluster(peerCluster ramen.DRCluster) (bool, error) {
+	u.log.Info(fmt.Sprintf("cleaning the cluster fence resource from the cluster %s", peerCluster.Name))
+	// TODO: delete the fencing CR MW.
+
+	err := u.mwUtil.DeleteManifestWork(fmt.Sprintf(util.ManifestWorkNameFormat,
+		u.object.Name, peerCluster.Name, util.MWTypeNF), peerCluster.Name)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			setDRClusterCleanCondition(&u.object.Status.Conditions, u.object.Generation, "fencing resource cleaned from cluster")
+
+			return false, nil
+		}
+
+		setDRClusterCleaningFailedCondition(&u.object.Status.Conditions, u.object.Generation,
+			"failed to clean fencing resource")
+
+		return true, fmt.Errorf("failed to create NetworkFence resource from cluster %s", peerCluster.Name)
+	}
+
+	setDRClusterCleaningCondition(&u.object.Status.Conditions, u.object.Generation, "NetworkFence resource clean started")
+
+	// Since ManifestWork for the fencing CR delete request
+	// has been just issued, requeue is needed to ensure that
+	// the fencing CR has indeed been deleted from the cluster.
+	return true, nil
+}
+
+func getPeerCluster(ctx context.Context, list ramen.DRPolicyList, reconciler *DRClusterReconciler,
+	object *ramen.DRCluster, log logr.Logger) (ramen.DRCluster, error) {
+	var peerCluster ramen.DRCluster
+
+	found := false
+
+	log.Info(fmt.Sprintf("number of DRPolicies found: %d", len(list.Items)))
+
+	for i := range list.Items {
+		drp := &list.Items[i]
+
+		log.Info(fmt.Sprintf("DRPolicy: %s, DRClusters: (%d) %v", drp.Name, len(drp.Spec.DRClusters),
+			drp.Spec.DRClusters))
+
+		// TODO: let policy = [e1, e2, e3]. Now, if e1 has to be fenced off,
+		//       it will be created on either of e2 or e3. And later when e1
+		//       has to be unfenced, the unfence should go to the same cluster
+		//       where fencing CR was created. For now, assumption is that
+		//       drPolicies will be having 2 clusters.
+		for _, cluster := range drp.Spec.DRClusters {
+			// skip if cluster is this drCluster
+			if cluster == object.Name {
+				drCluster, err := getPeerFromPolicy(ctx, reconciler, log, drp, object)
+				if err != nil {
+					log.Error(err, fmt.Sprintf("failed to get peer cluster for cluster %s", cluster))
+
+					break
+				}
+
+				peerCluster = *drCluster
+				found = true
+
+				break
+			}
+
+			if found {
```

Comment:
Shouldn't this be for the outer for loop?

Reply:
Thanks, good catch. I assumed it was in the outer for loop itself after coming out of the if loop.